GBA/core.zig

```diff
@@ -26,71 +20,53 @@ pub const GBA = struct {
     pub const MODE4_SCREEN_SIZE = 0x9600;
     pub const MODE5_SCREEN_SIZE = 40 * 1024;
 
-    pub const PaletteBank = [16]u16;
-
-    pub const PaletteMode = enum(u1) {
-        Color16,
-        Color256,
+    pub const Interrupt = enum {
+        VBlank,
+        HBlank,
+        Timer0,
+        Timer1,
+        Timer2,
+        Timer3,
+        Serial,
+        DMA0,
+        DMA1,
+        DMA2,
+        DMA3,
+        Keypad,
+        Gamepak,
     };
 
-    pub const InterruptFlags = packed struct {
-        vblank: bool = false,
-        hblank: bool = false,
-        vcounter: bool = false,
-        timer0: bool = false,
-        timer1: bool = false,
-        timer2: bool = false,
-        timer3: bool = false,
-        serial: bool = false,
-        dma0: bool = false,
-        dma1: bool = false,
-        dma2: bool = false,
-        dma3: bool = false,
-        keypad: bool = false,
-        gamepak: bool = false,
-    };
+    pub const InterruptFlags = std.EnumSet(Interrupt);
 
     pub const Header = extern struct {
-        romEntryPoint: u32 align(1),
-        nintendoLogo: [156]u8 align(1),
-        gameName: [12]u8 align(1),
-        gameCode: [4]u8 align(1),
-        makerCode: [2]u8 align(1),
-        fixedValue: u8 align(1),
-        mainUnitCode: u8 align(1),
-        deviceType: u8 align(1),
-        reservedArea: [7]u8 align(1),
-        softwareVersion: u8 align(1),
-        complementCheck: u8 align(1),
-        reservedArea2: [2]u8 align(1),
+        romEntryPoint: u32 align(1) = 0xEA00002E,
+        /// Game will not boot if these values are changed.
+        nintendoLogo: [156]u8 align(1) = .{
+            0x24, 0xFF, 0xAE, 0x51, 0x69, 0x9A, 0xA2, 0x21, 0x3D, 0x84, 0x82, 0x0A, 0x84, 0xE4, 0x09, 0xAD,
+            0x11, 0x24, 0x8B, 0x98, 0xC0, 0x81, 0x7F, 0x21, 0xA3, 0x52, 0xBE, 0x19, 0x93, 0x09, 0xCE, 0x20,
+            0x10, 0x46, 0x4A, 0x4A, 0xF8, 0x27, 0x31, 0xEC, 0x58, 0xC7, 0xE8, 0x33, 0x82, 0xE3, 0xCE, 0xBF,
+            0x85, 0xF4, 0xDF, 0x94, 0xCE, 0x4B, 0x09, 0xC1, 0x94, 0x56, 0x8A, 0xC0, 0x13, 0x72, 0xA7, 0xFC,
+            0x9F, 0x84, 0x4D, 0x73, 0xA3, 0xCA, 0x9A, 0x61, 0x58, 0x97, 0xA3, 0x27, 0xFC, 0x03, 0x98, 0x76,
+            0x23, 0x1D, 0xC7, 0x61, 0x03, 0x04, 0xAE, 0x56, 0xBF, 0x38, 0x84, 0x00, 0x40, 0xA7, 0x0E, 0xFD,
+            0xFF, 0x52, 0xFE, 0x03, 0x6F, 0x95, 0x30, 0xF1, 0x97, 0xFB, 0xC0, 0x85, 0x60, 0xD6, 0x80, 0x25,
+            0xA9, 0x63, 0xBE, 0x03, 0x01, 0x4E, 0x38, 0xE2, 0xF9, 0xA2, 0x34, 0xFF, 0xBB, 0x3E, 0x03, 0x44,
+            0x78, 0x00, 0x90, 0xCB, 0x88, 0x11, 0x3A, 0x94, 0x65, 0xC0, 0x7C, 0x63, 0x87, 0xF0, 0x3C, 0xAF,
+            0xD6, 0x25, 0xE4, 0x8B, 0x38, 0x0A, 0xAC, 0x72, 0x21, 0xD4, 0xF8, 0x07,
+        },
+        gameName: [12]u8 align(1) = [_]u8{0x00} ** 12,
```

Comment:
If you migrate to the newer mach nominated Zig version (2024.10.0), you can do this:

`game_name: [12]u8 align(1) = @splat(0),`